src/p2/document.rs

```diff
@@ -783,7 +801,11 @@ impl Document {
         match v {
             ftd::PropertyValue::Value { value, .. } => self.value_to_json(value),
             ftd::PropertyValue::Reference { name, .. } => self.json(name),
-            _ => unreachable!(),
+            ftd::PropertyValue::Variable { .. } => Err(ftd::p1::Error::TypeNotCovertible {
+                doc_id: "".to_string(),
+                line_number: 0,
```

Comment:
Pass these two parameters to `object_to_json()`. We always call this method from a processor and processor has section, which contains line number.